index.bs

```diff
@@ -385,59 +412,83 @@ To <dfn>process an attribution source</dfn> given an [=attribution source=] |sou
 1. [=list/Remove=] all entries in |cache| where the entry's [=attribution source/expiry=] value is less than the current time.
 1. If the [=list/size=] of |cache| is less than an implementation-defined limit, [=set/append=] |source| to |cache|.
 
+
+# Triggering Algorithms # {#trigger-algorithms}
+
 <h3 algorithm id="attribution-trigger-creation">Creating an attribution trigger</h3>
 
 To <dfn>obtain an attribution trigger</dfn> given a [=url=] |url| and an
-[=environment settings object=] |environment|, return a [=attribution trigger=]
-with the items:
+[=environment settings object=] |environment|, run the following steps:
+
+1. Let |triggerData| be 0.
+1. If |url|'s [=url/query=] has a `"data"` field, set |triggerData| to the result of running [=parse attribution data=] with
+    the value associated with field modulo the user agent's [=max trigger data value=].
+1. Let |trigger| be a new [=attribution trigger=] with the items:
 
     : [=attribution trigger/trigger origin=]
     :: |environment|'s [=environment/top-level origin=].
     : [=attribution trigger/trigger data=]
-    :: The result of applying [=parse attribution data=] with the value associated with the
-        `"data"` field of |url|'s [=url/query=] modulo the user agent's [=max trigger data value=].
+    :: |triggerData|
     : [=attribution trigger/trigger time=]
     :: The current time.
     : [=attribution trigger/reporting endpoint=]
     :: |url|'s [=url/origin=]
+1. Return |trigger|
 
 <dfn>Max trigger data value</dfn> is a vendor specific integer which controls the potential values of [=attribution report/trigger data=].
 
 Issue: Formalize how to parse the query similar to URLSearchParams.
 
-<h3 algorithm id="triggering-attribution">Triggering attribution</h3>
+<h3 dfn id="should-block-response">Should internalResponse to request be blocked as attribution trigger</h3>
 
-To <dfn>trigger attribution</dfn> from a [=request=] |request|, run the following steps:
+Given a [=request=] |request|:
```

Comment:
I think we need to check Permissions Policy here

Reply:
Added the permissions policy definition here and elsewhere.